src/molecules/advanced-calendar/advanced-calendar.stories.tsx

```diff
@@ -35,11 +35,116 @@ export default {
   },
 } as Meta<typeof AdvancedCalendar>;
 
-export const Template: StoryObj<typeof AdvancedCalendar> = {
+export const Default: StoryObj<typeof AdvancedCalendar> = {
+  args: {
+    presets: true,
+    view: 'single',
+    toYear: 2025,
+  },
+  render: (args) => {
+    const [filteredValue, setFilteredValue] = useState<string>('');
+    return (
+      <AdvancedCalendar
+        {...args}
+        initialFocus
+        mode="single"
+        onSelect={(value) => {
+          setFilteredValue(value?.toISOString() || '');
+        }}
+        selected={filteredValue ? new Date(filteredValue) : undefined}
+      />
+    );
+  },
+};
```

Comment:
we have the same code for 4 functions here, I think it can be better to have this as reuseable function instead of writing it again and again.